packages/remix-dev/config.ts

```diff
@@ -608,8 +608,39 @@ export async function readConfig(
     tsconfigPath = rootJsConfig;
   }
 
+  // Note: When a future flag is removed from here, it should be added to the
+  // list below so we can let folks know if they have obsolete flags in their
+  // config.  If we ever convert remix.config.js to a TS file so we get proper
+  // typings this won't be necessary anymore.
   let future: FutureConfig = {};
 
+  if (appConfig.future) {
+    let userFlags = appConfig.future;
+    let deprecatedFlags = [
+      "unstable_cssModules",
+      "unstable_cssSideEffectImports",
+      "unstable_dev",
+      "unstable_postcss",
+      "unstable_tailwind",
+      "unstable_vanillaExtract",
```

Comment:
We already warned about these in v1

```suggestion
```

Reply:
I was envisioning this as an ongoing list.  There's no guarantee that the v1 warning got addressed, so if it's still relevant in v2 shouldn't we still let them know?  And if it did get addressed the inclusion of it here is a no-op.

Reply:
Than this could become a really long list in time as it will always be "relevant"

I would say that if they didn't address the warnings in v1, they won't in v2 either 🤷‍♂️
Having these flags will also be a no-op though

Reply:
I still learn towards including them because it's server-side build-time code, doesn't bloat any bundles, and no-ops if unnecessary.  Some `unstable_` flags only got removed recently ([1.16.0](https://github.com/remix-run/remix/releases/tag/remix%401.16.0)) and there's no guarantee that someone installed a version between 1.16.0 and v2.  Lots of folks only do upgrades once or twice a year so it's not uncommon to upgrade through a bunch of releases (i.e., 1.12 -> v2) and never see the short-lived v1 warning.

It's not to say that we need to keep v2 flags in something like v5, but having them live on through at least one major upgrade seems prudent to me.  Then someone would have to go v1 -> v3 to miss out.